addons/mail/static/src/models/composer/composer.js

```diff
@@ -51,6 +51,7 @@ function factory(dependencies) {
          */
         closeSuggestions() {
             this.update({ suggestionDelimiterPosition: clear() });
+            this.suggestionList.closeSuggestions();
```

Comment:
Removing the delimiter should clear the suggestion search term which should make the suggestion list empty.
It shouldn't be needed to call any manual "close" on it.